src/main/java/uk/gemwire/waitress/authentication/Auth.java

```diff
@@ -1,93 +1,304 @@
-     *
+     * <p>
      * Uses a 2^12 (4096) iteration factor.
+     *
      * @param password the bytes of a password string to hash.
      * @return the bytes of a secure hashed password.
      */
     public static byte[] createPasswordHash(byte[] password) {
         return BCrypt.withDefaults().hash(12, password);
     }
 
+    /**
+     * TODO Switch to TOML in the future
+     * This should only be called once during setup, as it does not clear current data
+     * Password for tmvkrpxl0 account is "password"
+     *
+     * Currently, user file format is:
+     * ORGN `List of organizations`
+     * TEAM `List of teams, each team entry follows this form: team_name[organization_name]. Bracket [] is necessary`
+     *
+     * `user's name` `password hash` `List of organizations`
+     * `user's name` `password hash` `List of organizations`
+     * `user's name` `password hash` `List of organizations`
+     *
+     * Temporary user file is `data/users`
+     * @param userFile User file path
+     */
+    public static void loadUsers(final File userFile) throws IOException, CalledTooEarlyException {
+        try(final BufferedReader reader = new BufferedReader(new FileReader(userFile, StandardCharsets.UTF_8))) {
+            String firstLine = reader.readLine().substring(5); // First line is: "ORGN {List of organizations}", "ORGN " must be skipped
+            List<Organization> orgToAdd = Arrays.stream(firstLine.split(" ")).map(Organization::new).toList();
+            organizations.addAll(orgToAdd);
+
+            String secondLine = reader.readLine().substring(5);// Second line is: "TEAM {List of teams}", "TEAM " must be skipped
+            for (String teamEntry : secondLine.split(" ")) {
+                final int leftBracket = teamEntry.lastIndexOf('[');
+                final String orgName = teamEntry.substring(leftBracket + 1, teamEntry.length() - 1);
+
+                final String teamName = teamEntry.substring(0, leftBracket);
+                getOrganization(orgName).ifPresentOrElse(
+                        org -> {
+                            Team team = new Team(teamName, org);
+                            teams.add(team);
+                            org.addTeam(team);
+                        },
+                        () -> Waitress.LOGGER.warn("Organization " + orgName + " for team " + teamName + " does not exist! Ignoring..")
+                );
+            }
+
+            for (String userEntry : reader.lines().toList()) {// Rest of the file from now on is user entry
+                if (userEntry.isEmpty()) continue; // Ignore empty lines
+                final String[] split = userEntry.split(" ");
+                final String name = split[0];
+                final byte[] hash = split[1].getBytes(StandardCharsets.UTF_8);
+                final User user = addUser(name, hash); // Should it get existing user if user already exists?
```

Comment:
This method loads user data file, creates User instances, and configure their permissions. and it is expected to only run once during initialization.
As addUser returns null when user already exists, this will throw NPE when it's not called during initialization.
But should it just get existing user when one already exists? This allows configuring anonymous account's permission

Reply:
We should probably change this to getOrCreateUser or something like that, yeah.